tasks/Ubuntu.yml

```diff
@@ -0,0 +1,12 @@
+# Since Ubuntu's official package repositories **do not** support HTTPS
+# we cannot convert every URL to use HTTPS instead of HTTP. We must skip
+# any URLs that use the official repositories.
+- name: Convert any non-official HTTP source URLs to use HTTPS
+  ansible.builtin.replace:
+    path: "{{ item }}"
+    # Looking for entries per the format specified in
+    # https://manpages.ubuntu.com/manpages/jammy/man5/sources.list.5.html
+    # with a negative lookahead for URIs containing ubuntu.com or canonical.com
+    regexp: '(deb(?:-src)? (?:\[[^\]]+\] )?)http://(?!\S*(?:ubuntu|canonical)\.com)'
+    replace: '\1https://'
+  loop: "{{ apt_source_files }}"
```

Comment:
If you want, you could define the regex in the `vars` and avoid repeating what is almost the same task in two different files.  That seems DRYer to me.

Reply:
That's a good point. I only broke it up to make the documentation around was what being done clearer.

Reply:
The documentation could still live with the definition of the regex in `vars`.

Reply:
Please see [49e84ad](https://github.com/cisagov/ansible-role-apt-over-https/pull/3/commits/49e84ad29673684aeceb099a1a66aa4fbec6129b).

Reply:
Please see [here](https://github.com/cisagov/ansible-role-apt-over-https/pull/3#discussion_r855364729).